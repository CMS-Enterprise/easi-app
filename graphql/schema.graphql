type Document {
  id: ID!
  title: String!
  key: String!
  kind: String!
  project: Project!
}

enum DocumentType {
  VPAT
  DesignSessionDeck
  TRBLetter
}

type DocumentInput {
  title: String!
  key: String!
  type: DocumentType!
  project: Project!
}

type DocumentUpdateResponse {
  success: Boolean!
  message: String
  document: Document!
}

type Project {
  id: ID!
  name: String!
  documents: [Document]
}

type ProjectInput {
  name: String!
}

type ProjectUpdateResponse {
  success: Boolean!
  message: String
  project: Project!
}

type Query {
  projects: [Project]!
  project(id: ID!): Project
}

type Mutation {
  createProject(project: ProjectInput): ProjectUpdateResponse
  createDocument(input: DocumentInput): DocumentUpdateResponse
}
