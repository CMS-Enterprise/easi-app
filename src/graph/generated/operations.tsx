import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';

const defaultOptions = {};
export type GetSystemIntakeQueryVariables = Exact<{
  id: Scalars['UUID'];
}>;

export type GetSystemIntakeQuery = { __typename?: 'Query' } & {
  systemIntake?: Maybe<
    { __typename?: 'SystemIntake' } & Pick<
      SystemIntake,
      | 'id'
      | 'adminLead'
      | 'businessNeed'
      | 'businessSolution'
      | 'currentStage'
      | 'grbDate'
      | 'grtDate'
      | 'lcid'
      | 'needsEaSupport'
      | 'requestName'
      | 'requestType'
      | 'status'
      | 'submittedAt'
    > & {
        businessOwner?: Maybe<
          { __typename?: 'SystemIntakeBusinessOwner' } & Pick<
            SystemIntakeBusinessOwner,
            'component' | 'name'
          >
        >;
        contract?: Maybe<
          { __typename?: 'SystemIntakeContract' } & Pick<
            SystemIntakeContract,
            'contractor' | 'hasContract' | 'vehicle'
          > & {
              endDate?: Maybe<
                { __typename?: 'ContractDate' } & Pick<
                  ContractDate,
                  'day' | 'month' | 'year'
                >
              >;
              startDate?: Maybe<
                { __typename?: 'ContractDate' } & Pick<
                  ContractDate,
                  'day' | 'month' | 'year'
                >
              >;
            }
        >;
        costs?: Maybe<
          { __typename?: 'SystemIntakeCosts' } & Pick<
            SystemIntakeCosts,
            'isExpectingIncrease' | 'expectedIncreaseAmount'
          >
        >;
        grtFeedbacks: Array<
          { __typename?: 'GRTFeedback' } & Pick<
            GrtFeedback,
            'feedback' | 'feedbackType' | 'createdAt'
          >
        >;
        governanceTeams?: Maybe<
          { __typename?: 'SystemIntakeGovernanceTeam' } & Pick<
            SystemIntakeGovernanceTeam,
            'isPresent'
          > & {
              teams?: Maybe<
                Array<
                  { __typename?: 'SystemIntakeCollaborator' } & Pick<
                    SystemIntakeCollaborator,
                    'acronym' | 'collaborator' | 'key' | 'label' | 'name'
                  >
                >
              >;
            }
        >;
        isso?: Maybe<
          { __typename?: 'SystemIntakeISSO' } & Pick<
            SystemIntakeIsso,
            'isPresent' | 'name'
          >
        >;
        fundingSource?: Maybe<
          { __typename?: 'SystemIntakeFundingSource' } & Pick<
            SystemIntakeFundingSource,
            'fundingNumber' | 'isFunded' | 'source'
          >
        >;
        productManager?: Maybe<
          { __typename?: 'SystemIntakeProductManager' } & Pick<
            SystemIntakeProductManager,
            'component' | 'name'
          >
        >;
        requester: { __typename?: 'SystemIntakeRequester' } & Pick<
          SystemIntakeRequester,
          'component' | 'email' | 'name'
        >;
      }
  >;
};

export const GetSystemIntakeDocument = gql`
  query GetSystemIntake($id: UUID!) {
    systemIntake(id: $id) {
      id
      adminLead
      businessNeed
      businessSolution
      businessOwner {
        component
        name
      }
      contract {
        contractor
        endDate {
          day
          month
          year
        }
        hasContract
        startDate {
          day
          month
          year
        }
        vehicle
      }
      costs {
        isExpectingIncrease
        expectedIncreaseAmount
      }
      currentStage
      grbDate
      grtDate
      grtFeedbacks {
        feedback
        feedbackType
        createdAt
      }
      governanceTeams {
        isPresent
        teams {
          acronym
          collaborator
          key
          label
          name
        }
      }
      isso {
        isPresent
        name
      }
      fundingSource {
        fundingNumber
        isFunded
        source
      }
      lcid
      needsEaSupport
      productManager {
        component
        name
      }
      requester {
        component
        email
        name
      }
      requestName
      requestType
      status
      submittedAt
    }
  }
`;

/**
 * __useGetSystemIntakeQuery__
 *
 * To run a query within a React component, call `useGetSystemIntakeQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetSystemIntakeQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetSystemIntakeQuery({
 *   variables: {
 *      id: // value for 'id'
 *   },
 * });
 */
export function useGetSystemIntakeQuery(
  baseOptions: Apollo.QueryHookOptions<
    GetSystemIntakeQuery,
    GetSystemIntakeQueryVariables
  >
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useQuery<GetSystemIntakeQuery, GetSystemIntakeQueryVariables>(
    GetSystemIntakeDocument,
    options
  );
}
export function useGetSystemIntakeLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<
    GetSystemIntakeQuery,
    GetSystemIntakeQueryVariables
  >
) {
  const options = { ...defaultOptions, ...baseOptions };
  return Apollo.useLazyQuery<
    GetSystemIntakeQuery,
    GetSystemIntakeQueryVariables
  >(GetSystemIntakeDocument, options);
}
export type GetSystemIntakeQueryHookResult = ReturnType<
  typeof useGetSystemIntakeQuery
>;
export type GetSystemIntakeLazyQueryHookResult = ReturnType<
  typeof useGetSystemIntakeLazyQuery
>;
export type GetSystemIntakeQueryResult = Apollo.QueryResult<
  GetSystemIntakeQuery,
  GetSystemIntakeQueryVariables
>;
