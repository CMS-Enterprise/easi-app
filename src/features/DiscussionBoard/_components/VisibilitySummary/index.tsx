import React from 'react';
import { useTranslation } from 'react-i18next';
import { Icon, SummaryBox } from '@trussworks/react-uswds';
import { SystemIntakeGRBDiscussionBoardType } from 'gql/generated/graphql';

type VisibilitySummaryProps = {
  discussionBoardType: SystemIntakeGRBDiscussionBoardType;
};

/**
 * Summary box that lists the users who can view
 * and interact with a discussion board
 */
const VisibilitySummary = ({ discussionBoardType }: VisibilitySummaryProps) => {
  const { t } = useTranslation('discussions');

  return (
    <SummaryBox data-testid="visibility-summary">
      <p className="margin-0">
        {t('governanceReviewBoard.discussionsSharedWith')}
      </p>
      <ul className="usa-list usa-list--unstyled margin-top-1">
        {discussionBoardType === SystemIntakeGRBDiscussionBoardType.PRIMARY && (
          <li className="display-flex flex-align-center flex-column-gap-05">
            <Icon.Person aria-hidden />
            {t('tags.REQUESTER')}
          </li>
        )}

        <li className="display-flex flex-align-center flex-column-gap-05">
          <Icon.People aria-hidden />
          {t('tags.GROUP_GRB_REVIEWERS')}
        </li>
        <li className="display-flex flex-align-center flex-column-gap-05">
          <Icon.People aria-hidden />
          {t('tags.GROUP_IT_GOV')}
        </li>
      </ul>
    </SummaryBox>
  );
};

export default VisibilitySummary;
