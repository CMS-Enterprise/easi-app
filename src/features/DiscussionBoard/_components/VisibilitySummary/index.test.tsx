import React from 'react';
import { render, screen, within } from '@testing-library/react';
import { SystemIntakeGRBDiscussionBoardType } from 'gql/generated/graphql';

import VisibilitySummary from '.';

describe('VisibilitySummary', () => {
  it('renders the visibility summary for the Primary discussion board', () => {
    const { asFragment } = render(
      <VisibilitySummary
        discussionBoardType={SystemIntakeGRBDiscussionBoardType.PRIMARY}
      />
    );

    expect(asFragment()).toMatchSnapshot();

    const rolesList = within(
      screen.getByTestId('visibility-summary')
    ).getByRole('list');

    expect(
      within(rolesList)
        .getAllByRole('listitem')
        .find(item => item.textContent === 'Requester')
    ).toBeInTheDocument();

    expect(
      within(rolesList)
        .getAllByRole('listitem')
        .find(item => item.textContent === 'Governance Review Board (GRB)')
    ).toBeInTheDocument();

    expect(
      within(rolesList)
        .getAllByRole('listitem')
        .find(item => item.textContent === 'Governance Admin Team')
    ).toBeInTheDocument();
  });

  it('renders the visibility summary for the Internal discussion board', () => {
    render(
      <VisibilitySummary
        discussionBoardType={SystemIntakeGRBDiscussionBoardType.INTERNAL}
      />
    );

    const rolesList = within(
      screen.getByTestId('visibility-summary')
    ).getByRole('list');

    expect(
      within(rolesList)
        .getAllByRole('listitem')
        .find(item => item.textContent === 'Requester')
    ).toBeUndefined();
  });
});
