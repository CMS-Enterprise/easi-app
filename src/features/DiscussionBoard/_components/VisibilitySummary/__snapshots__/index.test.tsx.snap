// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VisibilitySummary > renders the visibility summary for the Primary discussion board 1`] = `
<DocumentFragment>
  <div
    class="usa-summary-box"
    data-testid="visibility-summary"
  >
    <div
      class="usa-summary-box__body"
    >
      <p
        class="margin-0"
      >
        All discussions and replies in this discussion board will be shared with:
      </p>
      <ul
        class="usa-list usa-list--unstyled margin-top-1"
      >
        <li
          class="display-flex flex-align-center flex-column-gap-05"
        >
          <svg
            aria-hidden="true"
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
          Requester
        </li>
        <li
          class="display-flex flex-align-center flex-column-gap-05"
        >
          <svg
            aria-hidden="true"
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
          Governance Review Board (GRB)
        </li>
        <li
          class="display-flex flex-align-center flex-column-gap-05"
        >
          <svg
            aria-hidden="true"
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
          Governance Admin Team
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
